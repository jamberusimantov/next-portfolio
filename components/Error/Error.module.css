.errorContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    /* min-height: calc(100vh - 5.5rem - 75px); */
}

.errorContainer h1 {
    line-height: 1.15;
    font-size: 4rem;
    text-align: center;
    color: var(--black);
}

.errorContainer h1 span {
    color: var(--blue);
}

.errorContainer h3 {
    text-align: center;
}

.errorContainer p {
    word-break: break-all;
    margin: 2rem 0;
    line-height: 1.5;
    font-size: 1rem;
    font-family: Verdana, Geneva, Tahoma, sans-serif;
    color: #eaeaea;
    background-color: var(--black);
    border: 1px solid var(--black-coal);
    border-radius: 10px;
    padding: 0 1.5rem;
    text-align: center;
}

.links {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.link {
    padding: .5rem 1rem;
    margin: 0 .5rem 1rem;
    border: 1px solid var(--blue);
    background-color: var(--white);
    cursor: pointer;
    display: inline-flex;
    justify-content: center;
    align-items: center;
    width: 150px;
    font-size: large;
    font-weight: 100;
    color: var(--blue);
}

.link:hover {
    color: var(--white);
    background-color: var(--blue);
}

@media(min-width:768px) {
    .errorContainer p {
        font-size: 1.5rem;
    }
}