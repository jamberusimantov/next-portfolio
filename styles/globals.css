html,
body {
    padding: 0;
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
    color: inherit;
    text-decoration: none;
}

* {
    box-sizing: border-box;
}

.hamburger {
    width: 40px;
    height: 40px;
    border: 0 solid transparent;
    border-radius: 0;
    cursor: pointer;
    display: inline-block;
    padding: 5px;
    background-color: transparent;
    opacity: 1;
}

.inner {
    margin: -1px 0 0;
    position: relative;
    width: 100%
}

.hamburger .bar {
    background-color: var(--white);
    border-radius: 0;
    content: "";
    display: block;
    height: 2px;
    position: absolute;
    width: 100%
}

.hamburger .bar:first-child {
    top: -10px;
}

.hamburger .bar:nth-child(3) {
    top: 10px;
}

.hamburger.active .bar:first-child {
    transform: rotate(45deg) translate3d(7px, 7px, 0);
    transition-delay: .1s
}

.hamburger.active .bar:nth-child(2) {
    transform: scaleX(0);
    transition-delay: .05s
}

.hamburger.active .bar:nth-child(3) {
    transform: rotate(-45deg) translate3d(7px, -7px, 0);
    transition-delay: .1s
}

.page {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    font-size: large;
}

.pageContainer {
    position: relative;
    min-height: calc(100vh - 4.5rem);
}

.submitBtn {
    width: 100%;
    max-width: 150px;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: .5rem;
}

.submitBtn p {
    flex: 1;
    margin: 0;
    font-size: large;
    padding: 0 !important;
}

.response {
    height: 25px;
    width: 25px;
    padding: 0 2rem;
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
}

.success {
    background-image: url('/assets/svg/success.svg');
}

.failure {
    background-image: url('/assets/svg/failure.svg');
}

.submit {
    background-image: url('/assets/svg/submit.svg');
}

@media(min-width:580px) {
    .submitBtn {
        margin: auto;
    }
}

@media(min-width:768px) {
    .pageContainer {
        min-height: calc(100vh - 7rem);
    }
}