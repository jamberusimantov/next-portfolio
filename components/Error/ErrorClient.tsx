import React from 'react'
import styles from './Error.module.css'
import { useRouter } from 'next/router'
import { CustomBtn } from '../../dir/elements'


const ErrorClient = () => {

    const router = useRouter();

    return (
        <div className={styles.errorContainer}>
            <h1 className={styles.title}>
                Oops... <span>Bad request!</span>
            </h1>
            <p className={styles.path}>
                <code>
                    https://portfoio-simantov.herokuapp.com{router.asPath}
                </code>
            </p>
            <CustomBtn
                click={() => { router.replace('/') }}
                className='btn'
                child={{
                    name: 'home',
                    size: 35
                }}
            />
        </div>
    )
}

export default ErrorClient;