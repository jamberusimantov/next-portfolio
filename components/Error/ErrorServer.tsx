import React from 'react'
import styles from './Error.module.css'
import { CustomLink } from '../../dir/elements'


const ErrorServer = () => {

    return (
        <div className={`${styles.container} ${styles.errorContainer}`}>

            <div className={styles.errorTitleContainer}>
                <h1 className={styles.title}>
                    Oops... <span>Server error!</span>
                </h1>
            </div>

            <h3 className={styles.recommendationsTitle}>
                Sorry for any inconvenience this may cause, here are some other Amazing recommendations
            </h3>

            <CustomLink
                url="https://memo-simantov.herokuapp.com/"
                className={styles.link}
                child={{
                    name: 'pencil',
                    text: 'Memo',
                    size: 35
                }}
            />

            <CustomLink
                url='https://weather-simantov.herokuapp.com/'
                className={styles.link}
                child={{
                    name: 'weather',
                    text: 'Weather',
                    size: 35
                }}
            />

            <CustomLink
                url='https://jamberusimantov.github.io/snake/'
                className={styles.link}
                child={{
                    name: 'game',
                    text: 'Snake',
                    size: 35
                }}
            />

        </div>
    )
}

export default ErrorServer;