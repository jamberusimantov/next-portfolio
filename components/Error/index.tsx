import React, { useContext } from 'react';
import styles from './Error.module.css';
import { useRouter } from 'next/router';
import CustomLink from '../CustomLink';
import dictionary from '../../dir/dictionary';
import HTMLContext from '../../contexts/htmlContext';


interface iErrorProps { pageName: keyof { "pageNotFound": "", "badRequest": "" } }

const Error = (props: iErrorProps) => {
    const { html } = useContext(HTMLContext);
    return (
        <div className={styles.errorContainer}>
            <h1><span>Oops</span>...{dictionary[props.pageName][html.lang]}</h1>
            {props.pageName == "pageNotFound"
                ? <p>https://portfoio-simantov.herokuapp.com{useRouter()?.asPath}</p>
                : <h3>Sorry for any inconvenience this may cause</h3>
            }
            <div className={styles.links}>
                <CustomLink
                    href='/'
                    title='Homepage'
                    children={<div className={styles.link}>Homepage</div>}
                />
            </div>
        </div>
    );
};

export default Error;