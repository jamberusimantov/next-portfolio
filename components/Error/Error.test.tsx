/**
 * @jest-environment jsdom
 */
import React from "react";
import { describe, expect, test } from '@jest/globals';
import { render, act } from '@testing-library/react';
import Error from '.';


describe('Error', () => {
    test('use jsdom in this test file', () => {
        expect(document.createElement('div')).not.toBeNull();
    });
    test('renders correctly', async () => {
        let el: HTMLElement = document.createElement("div");
        let getText: Function = () => { };
        await act(async () => {
            let { container, getByText } = await render(<Error pageName="pageNotFound" />);
            el = container;
            getText = getByText;
        });
        expect(el).toMatchSnapshot();
        expect(getText("Oops"));
    });
});