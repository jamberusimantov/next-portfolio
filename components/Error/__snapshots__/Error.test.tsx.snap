// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<div>
  <div
    class="errorContainer"
  >
    <h1>
      <span>
        Oops
      </span>
      ...
      דף לא נמצא
    </h1>
    <p>
      portfoio-simantov.herokuapp.com
    </p>
    <div
      class="links"
    >
      <a
        class="link"
        href="/"
        title="Homepage"
      >
        <div
          class="link"
        >
          Homepage
        </div>
      </a>
    </div>
  </div>
</div>
`;
